src/commands/wallet.rs

```diff
@@ -1,21 +1,27 @@
-use std::str::FromStr;
-
-use crate::{
-    config::ZKSyncConfig,
-    utils::balance::{display_l1_balance, display_l2_balance},
-};
+use crate::config::ZKSyncConfig;
+use crate::utils::balance::display_balance;
+use crate::utils::wallet::get_wallet_l1_l2_providers;
 use clap::Subcommand;
 use eyre::ContextCompat;
+use spinoff::{spinner, spinners, Color, Spinner};
 use zksync_ethers_rs::{
     abi::Hash,
-    middleware::SignerMiddleware,
-    providers::{Middleware, Provider},
-    signers::{LocalWallet, Signer},
+    core::utils::{parse_ether, ConversionError},
     types::{Address, U256},
-    zk_wallet::ZKWallet,
-    ZKMiddleware,
+    wait_for_finalize_withdrawal, ZKMiddleware,
 };
 
+trait U256Parsed {
+    fn from_dec_str_and_parse(value: &str) -> Result<U256, ConversionError>;
+}
+
+impl U256Parsed for U256 {
+    fn from_dec_str_and_parse(value: &str) -> Result<U256, ConversionError> {
+        // Assuming all tokens have 18 decimals
+        parse_ether(value)
+    }
+}
+
```

Comment:
If i use `parse_ether` as is i get the following:
```
error: implementation of `std::ops::FnOnce` is not general enough
  --> src/commands/wallet.rs:25:10
   |
25 | #[derive(Subcommand, PartialEq)]
   |          ^^^^^^^^^^ implementation of `std::ops::FnOnce` is not general enough
   |
   = note: `fn(&'2 str) -> std::result::Result<zksync_ethers_rs::types::U256, zksync_ethers_rs::ethers_core::utils::ConversionError> {zksync_ethers_rs::ethers_core::utils::parse_ether::<&'2 str>}` must implement `std::ops::FnOnce<(&'1 str,)>`, for any lifetime `'1`...
   = note: ...but it actually implements `std::ops::FnOnce<(&'2 str,)>`, for some specific lifetime `'2`
   = note: this error originates in the derive macro `Subcommand` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
I'll look into this

Reply:
We can use
```
#[clap(long = "amount", value_parser = |f: &str| parse_ether(f))]
amount: U256,
```
but as it will appear multiple times I'd suggest using the trait's function not as a trait implementation to the type but as an regular function.

Reply:
Nice TY. Pushed, it's a simpler solution.